core/advancedpolls.php

```diff
@@ -346,6 +352,23 @@ public function do_poll_modification($topic_data)
 	}
 
 	/**
+	 * Add the possible options to the template
+	 *
+	 * @param array	$post_data		The array of post data
+	 * @return void
+	 */
+	public function do_poll_block_modification($poll_options_template_data)
+	{
+		usort($poll_options_template_data, array($this, "order_by_votes"));
+		return $poll_options_template_data;
+	}
+
+	protected function order_by_votes($a, $b)
+	{
+		return (((int) $a['POLL_OPTION_RESULT'] < (int) $b['POLL_OPTION_RESULT']) ? 1 : (((int) $a['POLL_OPTION_RESULT'] > (int) $b['POLL_OPTION_RESULT']) ? -1 : (((int) $a['POLL_OPTION_ID'] > (int) $b['POLL_OPTION_ID']) ? 1 : -1))); 
```

Comment:
Ah I see. usort is not stable, that's the problem.

In this case, you can use
```php
return ((int) $b['POLL_OPTION_RESULT'] - (int) $a['POLL_OPTION_RESULT'])) ?: ((int) $a['POLL_OPTION_ID'] - (int) $b['POLL_OPTION_ID']));
```
This is the shorthand [ternary operator](http://php.net/manual/en/language.operators.comparison.php#language.operators.comparison.ternary). See [here on stackoverflow](http://stackoverflow.com/questions/1993409/operator-the-elvis-operator-in-php) for an explanation.

But I would guess in this case I would go another way. Make a real if/else statement. That will be better readable.